File: src/main/java/com/funstat/sequenta/Signal.java
package com.funstat.sequenta;

public class Signal {
    public SignalType type;
    public Sequenta.Setup reference;
    Sequenta.Setup successor;

    public Signal(SignalType type, Sequenta.Setup reference) {
        this.type = type;
        this.reference = reference;
    }

    public Signal(SignalType type, Sequenta.Setup reference, Sequenta.Setup successor) {
        this.type = type;
        this.reference = reference;
        this.successor = successor;
    }

    @Override
    public String toString() {
        return "Signal{" +
                "type=" + type +
                '}';
    }
}

File: back/src/main/java/firelib/indicators/sequenta/Signal.kt
package firelib.indicators.sequenta

data class Signal(val type: SignalType, val reference: Sequenta.Setup, val successor: Sequenta.Setup? = null) {

    val idx = reference.data().size - 1

    override fun toString(): String {
        return "Signal{" +
                "type=" + type +
                '}'.toString()
    }
}
